core/tests/test_views.py

```diff
@@ -69,3 +84,35 @@ def test_add_post_unauthorized(client: Client, community: Community) -> None:
     response = client.post(reverse("post-create"), data=data)
     assert response.status_code == 302
     assert reverse("login") in response.url
+
+
+def test_post_delete(client: Client, create_post: Post) -> None:
+    post = create_post
+    delete_url = reverse("post-delete", args=[post.id])
+    post_detail_url = reverse("post-detail", args=[post.id])
+    post_list_url = reverse("post-list")
+
+    response = client.get(post_detail_url)
+    assert response.status_code == 200
+    assert post.title in str(response.content)
+    assert "Test Post Title" in str(response.content)
+
+    response = client.get(post_list_url)
+    assert response.status_code == 200
+    assert post.title in str(response.content)
+    assert "Test Post Title" in str(response.content)
+
+    response = client.post(delete_url, follow=True)
+    assert response.status_code == 200
+
+    assert response.redirect_chain[0][0] == post_list_url
+
+    response = client.get(post_detail_url)
+    assert response.status_code == 404
+
+    response = client.get(post_list_url)
+    assert response.status_code == 200
+    assert post.title not in str(response.content)
+    assert "Test Post Titlex" not in str(response.content)
+
+    assert not Post.objects.filter(id=post.id).exists()
```

Comment:
i would add another tests to check if other user can delete others posts

Reply:
Great point! However, let's handle this in a separate task to maintain the consistency of our community's policies. Could you please create a follow-up ticket for it?